python/ray/serve/_private/router.py

```diff
@@ -355,19 +359,23 @@ def __init__(
         deployment_id: DeploymentID,
         prefer_local_routing: bool = False,
         self_node_id: Optional[str] = None,
+        self_availability_zone: Optional[str] = None,
     ):
         self._loop = event_loop
         self._deployment_id = deployment_id
         self._prefer_local_routing = prefer_local_routing
         self._self_node_id = self_node_id
+        self._self_availability_zone = self_availability_zone
 
         # Current replicas available to be scheduled.
         # Updated via `update_replicas`.
         self._replica_id_set: Set[str] = set()
         self._replicas: Dict[str, ReplicaWrapper] = {}
         self._replicas_updated_event = make_asyncio_event_version_compat(event_loop)
         # Replicas colocated on same node
-        self._colocated_replica_ids: Set[str] = set()
+        self._node_colocated_replica_ids: Set[str] = set()
+        # Replicas colocated in same availability zone
+        self._az_colocated_replica_ids: Set[str] = set()
```

Comment:
Rather than having N fields for each locality scope, i'd suggest to:

 - Create an enum (having node, AZ for now) that we can extend in the future
 - Keep colocated replicas as map from enum -> set

Reply:
Thanks for the suggestion, implemented!